### PR TITLE
In 139255433 add allegations to screening

### PR DESCRIPTION
**Pull request by saraOSI**

### CODE REVIEW

File: spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx [hunk 1]
```diff
@@ -493,8 +493,67 @@ describe('ScreeningEditPage', () => {
...
+          perpetrator_id: perpetrator.id,
+          victim,
+          victim_id: victim.id,
+        }, {
+          id: null,
+          screening_id: '3',
+          perpetrator: anotherPerpetrator,
+          perpetrator_id: anotherPerpetrator.id,
+          victim,
+          victim_id: victim.id,
```
Comment (brunoOSI) [inline spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx]: why are we using `victim` and `victim_id` ? can't we just use the contents of `victim`? same for `perpetrator`

File: spec/javascripts/components/screenings/AllegationsShowViewSpec.jsx [hunk 2]
```diff
@@ -1,19 +1,64 @@
...
+
   it('renders allegations show view headings', () => {
-    const component = shallow(<AllegationsShowView onEdit={() => {}}/>)
+    const component = shallow(<AllegationsShowView {...requiredProps} />)
     expect(component.text()).toContain('Alleged Victim/Children')
     expect(component.text()).toContain('Alleged Perpetrator')
     expect(component.text()).toContain('Allegation(s)')
   })
 
+  it('renders alleged vicitms/perpetrators', () => {
```
Comment (brunoOSI) [inline spec/javascripts/components/screenings/AllegationsShowViewSpec.jsx]: victims typo here

File: spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx [hunk 3]
```diff
@@ -206,32 +205,6 @@ describe('ScreeningEditPage', () => {
...
-    it('calls screening save', () => {
-      expect(saveScreening).toHaveBeenCalledWith({name: 'mock screening'})
-    })
-
-    it('pushes show page to browser history', () => {
-      expect(browserHistory.push).toHaveBeenCalledWith({pathname: '/screenings/1'})
-    })
-  })
-
   describe('cardSave', () => {
```
Comment (brunoOSI) [inline spec/javascripts/components/screenings/ScreeningEditPageSpec.jsx]: 👍 yay deletes

File: spec/features/screening/allegations/edit_allegations_spec.rb [hunk 4]
```diff
@@ -193,4 +193,28 @@
...
+
+    within '#screening-information-card.card.edit' do
+      fill_in 'Title/Name of Screening', with: 'Hello'
+      click_button 'Save'
+    end
+
+    expect(
+      a_request(:put, api_screening_path(screening.id))
+      .with(json_body(remove_root_id(screening.as_json).merge('participants' => [])))
+    ).to have_been_made
```
Comment (brunoOSI) [inline spec/features/screening/allegations/edit_allegations_spec.rb]: shouldn't this test try to see if screening is equal to the put response ? I don't see a clear expectation here.

File: spec/features/screening/allegations/edit_allegations_spec.rb [hunk 5]
```diff
@@ -80,20 +168,24 @@
       expect(page).to have_no_selector('td')
     end
 
+      lisa.roles = ['Victim']
```
Comment (brunoOSI) [inline spec/features/screening/allegations/edit_allegations_spec.rb]: intentation is wrong here

File: spec/features/screening/participant/editing_participant_spec.rb [hunk 6]
```diff
@@ -273,7 +273,7 @@
 
     expect(
       a_request(:put, api_participant_path(marge.id))
-      .with(json_body(marge.to_h.tap { |h| h.delete(:id) }.as_json))
+      .with(json_body(remove_root_id(marge.as_json)))
```
Comment (saraOSI) [inline spec/features/screening/participant/editing_participant_spec.rb]: We sure can!

File: spec/features/screening/screening_information_spec.rb [hunk 7]
```diff
@@ -49,12 +49,12 @@
...
 
     stub_request(:put, api_screening_path(screening.id))
-      .with(json_body(screening.to_json(except: :id)))
+      .with(json_body(remove_root_id(screening.as_json)))
       .and_return(json_body(screening.to_json))
 
     expect(
       a_request(:put, api_screening_path(screening.id))
-      .with(json_body(screening.to_json(except: :id)))
+      .with(json_body(remove_root_id(screening.as_json)))
```
Comment (saraOSI) [inline spec/features/screening/screening_information_spec.rb]: Fixed per comment above.

File: spec/features/screening/allegations/edit_allegations_spec.rb (hunk 5 above)
Comment (saraOSI) [inline spec/features/screening/allegations/edit_allegations_spec.rb]: Will fix.

File: spec/features/screening/allegations/edit_allegations_spec.rb (hunk 4 above)
Comment (saraOSI) [inline spec/features/screening/allegations/edit_allegations_spec.rb]: We're focused on testing the API here, so the expectation is focused on the response from the API. (Chatted offline about this with @brunoOSI and we're ok with leaving as-is.)

**Pull request merged**